src/actions/actions.js

```diff
@@ -285,15 +282,19 @@ export function login(successCallback, failCallback) {
     }
   }
 }
+
+export function logoutSuccess(dispatch) {
+  dispatch({ type: 'LOGOUT' })
+  dispatch(updateAppMessages({ message: 'Logged Out' }))
+}
+
+export function logoutFailure(dispatch) {
+  dispatch(updateAppMessages({ message: 'Logout Failed' }))
+}
+
 export function logout() {
   return (dispatch) => {
-    fetch('/logout/')
-      .then((response) => {
-        if (response.ok) {
-          dispatch({ type: 'LOGOUT' })
-          dispatch(updateAppMessages({ message: 'Logged Out' }))
-        } else dispatch(updateAppMessages({ message: 'Logout Failed' }))
-      })
+    Logout(logoutSuccess(dispatch), logoutFailure(dispatch))
```

Comment:
I'm pretty sure this doesn't work, it looks like you're actually calling the functions here. Instead, we want to pass a reference to the functions so the logout function can call them itself.

Reply:
I am also thinking of . Somehow it doesn't give me error while I tested in my local environment . I should change this.

Reply:
@wlach I amend some changes regarding this can you look into it ?